x-pack/plugins/cases/public/components/custom_fields/types.ts

```diff
@@ -30,6 +36,7 @@ export interface CustomFieldType<T extends CaseUICustomField> {
 export type CustomFieldFactory<T extends CaseUICustomField> = () => {
   id: string;
   label: string;
+  getEuiTableColumn: (params: { key: string; label: string }) => EuiTableComputedColumnType<CaseUI>;
```

Comment:
I am a bit sceptical with this. Custom fields now have knowledge to the structure of cases which they shouldn't. What about we pass only the custom field to the render method? Example: `getEuiTableColumn: (params: { key: string; label: string }) => EuiTableComputedColumnType<CaseUICustomField>;` This way custom fields do not have to search on the case and are not aware of the structure. We can talk offline if you want of how to do it.

Reply:
I can pass `customField` as the field in the column and the render function will get the `customFields` array as a parameter, but I don't think we can get out of iterating over that array. Is this what you mean?

Reply:
I think we can. Let's talk about it offline.

Reply:
Discussed offline, found a solution.